### PR TITLE
post comments

### PR DESCRIPTION
**Pull request by eliotsnowdon**

### CODE REVIEW

File: src/Components/AddComment.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+import { useState } from "react"
+import { postComment } from "../apis"
+
+export const AddComment = ({setComment, review_id}) => {
+    const [commentChange, setCommentChange] = useState("");
+    console.log(commentChange)
```
Comment (RichardNeat) [inline src/Components/AddComment.jsx]: please remove any stray console logs

File: src/Components/AddComment.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        e.preventDefault();
+        const comment = {
+        "comment_id": Date.now(),
+        "body": commentChange,
+        "review_id": review_id,
+        "author": "jessjelly",
+        "votes": 0,
+        "created_at": "just now"
+        }
+        setComment((currComments) => {
```
Comment (RichardNeat) [inline src/Components/AddComment.jsx:21]: Nice work with the optimistic rendering

File: src/Components/AddComment.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            return [comment, ...currComments]
+        })
+        postComment(review_id, commentChange).then(() => {
+            setCommentChange("")
+        }).catch((err) => {
+            setComment((currComments)=> {
```
Comment (RichardNeat) [inline src/Components/AddComment.jsx:31]: I like this a lot, making sure that comment is removed if there is an error

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice work Eliot, we're really close. Just a couple of suggested changes:
- We are getting some errors in the console on the browser - can we have a look at those and fix.
- Remove console.log
- It would be nice to give some feedback to the user based on whether the comment was posted successfully or not. Consider that if we provide an error message, it should disappear if the user tries again.
- Currently I can leave a blank comment, I imagine this is behaviour that we would want to avoid. Consider disabling the button or some other form of feedback if the comment box is empty.
- Can we be sure that if the user is on a slow connection and they click submit that if that takes a long time they can't keep trying to submit the same comment?

Have a look at those points and give them a go, ask for help if you need, then resubmit when ready :)

**Comment by Dutton1990** [comment]
Nice job on not being able to post an empty comment, but I think we're still missing two of the things Rich mentioned above. 

From the ticket - What feedback are you giving to the user to show them their post has been successful?
You could insert a p tag based on some state that is changed if the post works, or something like a pop up, to tell the user that the post was successful. 

Also from the ticket - The post to the BE may take some time. How can you ensure the the user doesn't try and post the comment again?
You could disable the button once you press it. On refresh you'd be able to post another comment which is fine but you won't be able to spam the add button like I can do currently

**Review by RichardNeat — APPROVED** [review]
Nice one Eliot, you are welcome to go ahead and merge.
However please do look at your browser console before doing so, you have some errors popping up relating to keys, please have a look and fix before merging, when it's done go ahead.

**Pull request merged**